### ISSUE TITLE
Art: Main character soul seeds

### ISSUE THREAD

**Issue by madjin** [webaverse/app#1404]
Derived from issue https://github.com/webaverse/app/issues/1401#issuecomment-890045223, these are advanced technology vessels for AI souls that are captured in a non-human thing. They can be likened to a Navi from LOZ or Siri / Cortana / Echo in terms of artificial assistants. Can think of it like a USB x Drone technology in 20 years. 

![image](https://user-images.githubusercontent.com/32600939/128247937-bbe19463-6964-407e-b123-adfab1b93104.png)

Note: I like how the left image is like a 0x shape.

Similar to how Navi sometimes disappears into Link and reappears later, these seeds have a sleep function  / can idle as a wearable, such as being perched on avatar like parrot on a pirate's shoulder.

Ideas for the perched state:

hairclip / brooch / pin / patch
shoulder perched
watch / wristband
hat / scouter / headphones

Need to make 1 for each character, and from that we can figure out the right traits / properties for derivatives.

**Comment by madjin** [comment]
Name pending in terms of soul seeds.

**Comment by madjin** [comment]
Personally love this one, reminds me of a solarpunk 8ball droid

![image](https://user-images.githubusercontent.com/32600939/128271546-4e139bfd-f4f5-4466-8977-46868ca0eaf6.png)

This is how I imagine the droid to look, with wings (0x), like Navi

![Screenshot at 2021-08-04 20-01-04](https://user-images.githubusercontent.com/32600939/128271466-c5222697-701e-4169-9bd1-cb136ded0430.png)

Various designs

![Screenshot at 2021-08-04 20-11-09](https://user-images.githubusercontent.com/32600939/128271496-6949d757-cc0d-45a1-9f3a-032346bc3a5f.png)

![Screenshot at 2021-08-04 20-10-10](https://user-images.githubusercontent.com/32600939/128271484-e97c553f-0c77-4461-a2b8-bcae076c71fb.png)

These are digieggs: https://digimon.fandom.com/wiki/Digi-Egg_(Armor)

![image](https://user-images.githubusercontent.com/32600939/128271514-f288dc7c-7063-41ed-8543-898d908025f9.png)

**Comment by madjin** [comment]
Example of a perched soul attached to the avatar as a wearable. Don't need fan blades.

![](https://media.wired.com/photos/59323ff25c4fbd732b5516f7/master/w_2560%2Cc_limit/nixie-flying-gif.gif)

**Comment by GaladWarder** [comment]
Do these need a concept pass? And if so, what's the priority level? I see that the souls are backburnered so I just wanted to check.

**Comment by avaer** [comment]
We actually have a few people trying their art already, it is not backburnered.

But for the teaser we only need one of these for the main characters: https://github.com/webaverse/app/issues/1407

**Comment by avaer** [comment]
We still need a bunch of procedurally generated seeds -- it is not clear if we will start from 2D art or 3D art, it's whichever resonates best.

Some more concepts could be cool, but this is more a project for someone to take on the larger project of generating these variations, rather than concepting a couple.

**Comment by GaladWarder** [comment]
Makes sense, I'll leave this one be for now then!

**Comment by madjin** [comment]
Idea for when soul seeds germinate in darker areas of the great filters

![image](https://user-images.githubusercontent.com/32600939/129087635-7348e718-5df4-4d9d-abc4-57fe591c8fab.png)
![image](https://user-images.githubusercontent.com/32600939/129087622-f24b043d-6bf1-4f3a-808c-48daeeceac3e.png)
![image](https://user-images.githubusercontent.com/32600939/129087539-2f77fcf2-9444-49df-8438-0bf3b20baf37.png)
https://artsammich.blogspot.com/2015/02/re-post-of-infinity-incredibles.html
https://blenderartists.org/t/omnidroid-rigging-animation-practice/466563

**Comment by madjin** [comment]
new body shells from dreamco
![image](https://user-images.githubusercontent.com/32600939/129368099-67e2e4b7-7848-4972-a7cb-3a4c879aa249.png)

**Comment by madjin** [comment]
With faces
![image](https://user-images.githubusercontent.com/32600939/129424844-0693d9aa-a131-4e64-b509-4dd326cd9f56.png)

**Comment by madjin** [comment]
Inspiration for the next piece inspired by https://www.artstation.com/artwork/81mWw

![image](https://user-images.githubusercontent.com/32600939/129455660-fc0eca87-f2fd-4fde-98d3-805b72a832c5.png)

Wings from 
![image](https://user-images.githubusercontent.com/32600939/129455673-d5509696-6cbf-41f1-b747-9583f7d649a8.png)

places we can various traits (diff ears, legs, wings, arms, faces, etc)

- ears
  - cat, bear, dog, horns, antenna
- face
  - diff eyes / emoji including stop, loading, time
- wings / body texture
  - different metals, leafs, fairy wings, angel
   
![PureRef_7p18x6fv87](https://user-images.githubusercontent.com/32600939/129455687-631329b2-f310-4274-a4db-6712e93ef7f9.png)

**Comment by GaladWarder** [comment]
I'm not sure where these cross over, but I have done an ortho pass on the Morphers, which seem to also be the soul seeds? Going to crosslink here just in case it's relevant; I know the pass I did was for procgen.

https://github.com/webaverse/app/issues/1407#issuecomment-899983276

**Comment by madjin** [comment]
New shells by Dreamco

![image](https://user-images.githubusercontent.com/32600939/129911861-319e9d05-c440-49a9-a2d9-43a749d523b3.png)

**Comment by madjin** [comment]
Closing issue https://github.com/webaverse/app/issues/1407 in favor of this one

## V1

![](https://user-images.githubusercontent.com/29031531/128622035-01a5f254-75e5-451e-96c5-ab9e311c3061.jpg)

## V2

![](https://user-images.githubusercontent.com/29031531/128962617-a91dbb89-0ede-44c2-869c-431214f4972f.jpg)

![](https://user-images.githubusercontent.com/29031531/128962679-1a018032-4796-44d3-8d3e-ae89b8595427.jpg)

![](https://user-images.githubusercontent.com/29031531/129663123-1a0e6da5-26b6-4f0d-90a6-f34d33a1186a.jpg)

![](https://user-images.githubusercontent.com/29031531/129663454-30c1f2dd-de02-47e1-9260-fbdd2ad2113e.jpg)

**Comment by madjin** [comment]
Latest animation test of the wings
https://user-images.githubusercontent.com/32600939/129979120-dee11659-0882-4769-96db-67680c2b0eb5.mp4

To-do:

- [ ] UV faces
- [ ] Test swapping shells (get from dreamco)
- [ ] Slight bobbing float animation

This will provide enough for a MVP

**Comment by madjin** [comment]
base colors added
![image](https://user-images.githubusercontent.com/32600939/130073484-96be9ed8-b309-466e-ab28-6afe2e2a9574.png)

**Comment by plankatron** [comment]
https://user-images.githubusercontent.com/5592454/130133606-0fba0c70-8f18-445e-993c-929b47dbca34.mp4

**Comment by plankatron** [comment]
https://user-images.githubusercontent.com/5592454/130134832-80e50642-1ff1-4872-ab51-8eb7458ead9d.mp4

**Comment by madjin** [comment]
Modular components 3D modeled by Dreamco
![image](https://user-images.githubusercontent.com/32600939/130247070-38a89238-b5e8-436f-bff0-69fa7c48f078.png)

**Comment by GaladWarder** [comment]
Putting this here for posterity: https://github.com/webaverse/inspo/tree/main/items - the Morpher Pureref is in this tree

**Comment by GaladWarder** [comment]
Updating with notes from recent meeting regarding next concept pass:

- 3-4 control points for generative iterations: 1. wings, 2. core, 3. organic element, 4. texture/glow/materials/etc...
- Leaning more heavily into organic/solarpunk elements
- Middle of organic and tech, maybe a little magitech
- Possible to animate simply
- Smaller or different core-to-wing ratio (Could even potentially be a control point as well)

Sheets needed:
- Wing/core ratio iterations
- Silhouettes
- Rough sketches/linework/detail iteration
- Need to brainstorm the organic element control point

**Comment by GaladWarder** [comment]
Having a few other ideas as I work on silhouettes for this: for more control pointe, the relationships between the objects could also be a variable. For example, here is a simple shell with wings (navi style, just for demonstration purposes), in different relation to each other in space. In my opinion, it is enough of a difference to feel like a different iteration.

![image](https://user-images.githubusercontent.com/29031531/130372278-f3bdbd5a-50d3-41b3-8d06-d2e39bcf82f3.png)

If there was a value or set of values that were generated per iteration for things like wing distance from core, scaling on x, y, or z axis, rotation, or core size, it would give an extra layer of generative goodness with virtually no cost. Provided the values were within a range so they didn't break the mesh, they would mean that even two iterations that share all or most components would look distinct from each other. In a weird way, it would also allow for "twins" to be generated, which could increase the collectability of such a rare occurrence - two generations that share all components but have slightly different relation values.

**Comment by avaer** [comment]
Using placement as a control point makes sense to me.

**Comment by GaladWarder** [comment]
Putting this here; these are in no way final - just keeping everything together. Pushed and pulled some shapes and proportion to see what I could get away with, especially how we can get away from the 'navi' design. These are silhouette tests for the ratio of wings to core, with some super early shape ideas thrown in. Since we're taking it from the top, may as well start from total scratch.

![Morpher_V3_SilhouetteandProportionSheet](https://user-images.githubusercontent.com/29031531/130376445-0178c072-2b06-4ce4-8733-bd17787f7091.jpg)

Have also already started on a silhouette pass, getting more specific with some of the internal shapes especially. I'm overdoing the iterations in hopes that there will be multiple good designs that will come from it which can then become components.

**Comment by madjin** [comment]
Update on modular set for morphers, DreamCo is working on the UV layouts of the model next
![image](https://user-images.githubusercontent.com/32600939/130484183-1760b597-4da4-4757-8635-380fdf804f33.png)

Here are the other variants, pretty much ready to connect together if we want to start testing.
![image](https://user-images.githubusercontent.com/32600939/130484200-232c9e50-7acd-4bb7-b3b3-34798ba39acf.png)

Files have been uploaded to the drive.

**Comment by plankatron** [comment]
here is a test with dream's assets on the armature
![dream](https://user-images.githubusercontent.com/5592454/130515945-7f60701f-d5ec-4790-b1d4-4957dac6a66f.jpg)

**Comment by GaladWarder** [comment]
Been fiddling with some ideas more, I think I've found a pretty solid wing-to-body ratio and ideas and now I'm running through some iterations to try to nail down the general style/look. 

Pulling from the discussion yesterday, these are much more organic, less humanoid, and lean much harder into the solarpunk/magitech vibes.

Many of these also have glass showing their internals, which could also be an easily added control point - if there was a set of 'elements' like a bit of flame, blob of Silk, flower, smoke, etc... that were generated inside, similar to what Via did with Scilly's sword. Could even have Lisk bits in them - whatever fits the script better.

![Morpher_OrganicandMagic_V3-1](https://user-images.githubusercontent.com/29031531/130530821-cd4ec7d8-ee57-4eee-bb97-781ce391afd5.jpg)

I plan to do more of these, but wanted to get feedback first before I continue sinking time into it. Are these in the direction you had in mind? 

The control points I am designing around currently are:
- Wing type
- Core type
- Internal element
- Spacing/proportions of the previous 3 points

**Comment by madjin** [comment]
Texta made some quick models of these designs 
![image](https://user-images.githubusercontent.com/32600939/130550201-55e68e1d-de93-482e-b9a7-03c5d463baab.png)

**Comment by madjin** [comment]
Via had an awesome suggestion yesterday for having audio reactive emissive to represent the morpher when it communicates.

**Comment by madjin** [comment]
UVs / materials done, these are ready to be setup inside blender

![image](https://user-images.githubusercontent.com/32600939/130626346-24280766-1c64-4f3d-bc48-4c253c56d189.png)

**Comment by madjin** [comment]
I cut up the UVmap into 4x4 until spiral eyes are redone (see TTGL for inspiration), I can provide UVmap for the one to be tweaked.

![uv01](https://user-images.githubusercontent.com/32600939/130636055-3bdb42a9-a2f4-4e86-881d-639c8436f011.png)

![face](https://user-images.githubusercontent.com/32600939/130636261-5f5d73a7-51d3-435d-a363-003f118b6514.png)

![uv00](https://user-images.githubusercontent.com/32600939/130636065-b00b3842-325c-42c3-81ca-b6a02742151b.png)

**Comment by madjin** [comment]
Was playing with the materials to get a sense of how different textures with these + the faces might look. Here are some samples I made in a relatively short time.
![test_morphers](https://user-images.githubusercontent.com/32600939/130647519-d9143fb2-f78f-405d-8d56-09a88d9ff87f.png)

**Comment by GaladWarder** [comment]
One more sheet of these, pushing the different elements you guys liked a bit farther to see where it 'breaks'. Also messed with using some overpainted neural stuff in these which worked really well!

There's a ton of flexibility in this format, I don't think it will be any problem to have a very wide pool within each component.

![Morpher_OrganicandMagic_V3-2](https://user-images.githubusercontent.com/29031531/130707979-2338049b-8f2c-433d-99dc-22141feaf5e8.jpg)

Started on the face redesigns too for the swirly eyes, won't take long and will push them through tomorrow. 

What other assets/concepts/iterations are needed on my end for the newer versions of the morphers / seeds? Plan to be working on the hero image thumbnails tomorrow, but can get other stuff needed for this done fairly easily, too.

**Comment by plankatron** [comment]
https://user-images.githubusercontent.com/5592454/130709859-c0e2e2b4-e62a-407e-910a-5b12af7f9b53.mp4

**Comment by madjin** [comment]
Updated texture from Galad added to the wings. Test glb uploaded to drive (testy-mc-anim2.glb). 
https://user-images.githubusercontent.com/32600939/131204978-55ff0f8e-ca02-4786-90b2-1b0a657b3569.mp4

The filesize is good (2.3mb), but I think the polygon count can be much more reduced still (at ~25k atm)
![image](https://user-images.githubusercontent.com/32600939/131205049-8ee5f1f1-b320-40a3-bf6a-a24c8a48a94a.png)

**Comment by madjin** [comment]
All of the models have been organized into collections in a blend file

![blender_cphANgy27v](https://user-images.githubusercontent.com/32600939/131205301-70a4b59a-7f14-484c-8e5d-a90a51b723a1.png)

![brave_JaXlypimK1](https://user-images.githubusercontent.com/32600939/131205363-6d3abb77-76ad-419f-8208-edc548af1ff3.png)

@plankatron had to manually fix a bunch of vertex group issues:
- top most vert group not doing automatic weights well, needed to auto assign
- had to delete many extra vertex groups (automatic weights adds group for every bone)
- had to make sure vertex groups matched up, needed to manually move stuff around to fit right

Overall things work, the blend file is mostly ready to export sets from all the pieces DreamCo made. Here's a test (.glb uploaded to drive):
![de85083c97027d700217119b0ddabefc](https://user-images.githubusercontent.com/32600939/131205407-2da498cc-a23e-4218-920f-7b4dd6f6a470.gif)

I recorded a few videos with @plankatron on various aspects of the blend file to sync with devs on how it all works and what can be changed. Personally I think the wings need to be tweaked or perhaps added on later as an upgrade. This can be done non-destructively and I recorded a video as to how one can make variations. These droids look pretty good on their own while bobbing in the air even without wings.
